velox/connectors/hive/TableFormatReader.h

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/connectors/hive/HiveConnectorSplit.h"
+#include "velox/connectors/hive/TableHandle.h"
+#include "velox/dwio/common/BufferedInput.h"
+#include "velox/dwio/common/Reader.h"
+
+namespace facebook::velox::connector::hive {
+
+static const char* kPath = "$path";
+static const char* kBucket = "$bucket";
+
+class TableFormatReader {
```

Comment:
Let's expose `baseRowReader()` and `createRowReader` temporarily to allow Prism to migrate to use this

Reply:
ok. How does Prism uses these? To allow it to use them, I changed the baseRowReader_ from unique_ptr to shared_ptr. 
```
  std::unique_ptr<dwio::common::RowReader> createRowReader(
      dwio::common::RowReaderOptions& options) {
    return baseReader_->createRowReader(options);
  };

  std::shared_ptr<dwio::common::RowReader> baseRowReader() {
    return baseRowReader_;
  }
...
  std::unique_ptr<dwio::common::Reader> baseReader_;
  std::shared_ptr<dwio::common::RowReader> baseRowReader_;
```
Let me know if this is what you want. I think it's cleaner to make TableFormatReader to own these readers, but since you said it's temporary I assume these functions will be removed in near future?

Reply:
Thanks, `baseRowReader_` does not need to be shared; just make `dwio::common::RowReader& baseRowReader()` should be good.